### PR TITLE
update pci_hotplug cfg to contain the serial opton when hotplug block

### PR DESCRIPTION
**Pull request by PyLearner**
ID: 1340373

### CODE REVIEW

**Comment by PyLearner** [comment]
1. update the cfg command to get the hotplugged devname
2. update the hotplug command to contain the serial option

**Comment by ngu-niny** [comment]
After apply the patch, there is following issue:

03:44:38 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-37.0-py2.7.egg/avocado_vt/test.py:402
03:44:38 ERROR| Traceback (most recent call last):
03:44:38 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-37.0-py2.7.egg/avocado_vt/test.py", line 178, in runTest
03:44:38 ERROR|     self._runTest()
03:44:38 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-37.0-py2.7.egg/avocado_vt/test.py", line 320, in _runTest
03:44:38 ERROR|     run_func(self, params, env)
03:44:38 ERROR|   File "/usr/lib/python2.7/site-packages/autotest/client/shared/error.py", line 140, in new_fn
03:44:38 ERROR|     return fn(*args, **kwargs)
03:44:38 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/pci_hotplug.py", line 358, in run
03:44:38 ERROR|     add_device(pci_num, queues)
03:44:38 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/pci_hotplug.py", line 203, in add_device
03:44:38 ERROR|     after_add = add_fuction(pci_num, queues)
03:44:38 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/pci_hotplug.py", line 153, in device_add_block
03:44:38 ERROR|     pci_info[pci_num][0], serial_id))
03:44:38 ERROR| ValueError: incomplete format
03:44:38 ERROR|
03:44:38 ERROR| Traceback (most recent call last):

03:44:38 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-37.0-py2.7.egg/avocado_vt/test.py", line 416, in _run_avocado
    raise test_exception

03:44:38 ERROR| ValueError: incomplete format

03:44:38 ERROR| ERROR 1-smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_block.blk_virtio_blk.with_block_resize -> ValueError: incomplete format

File: qemu/tests/pci_hotplug.py [hunk 1]
```diff
@@ -142,13 +144,13 @@ def device_add_block(pci_num, queues=1):
 
         verify_supported_device(pci_model)
         if drive_cmd_type == "drive_add":
-            driver_add_cmd = ("%s auto file=%s,if=none,format=%s,id=%s" %
+            driver_add_cmd = ("%s auto file=%s,if=none,format=%s,id=%s,serial=%" %
```
Comment (ngu-niny) [inline qemu/tests/pci_hotplug.py]: It should be serial=%s.

**Comment by suqinhuang** [comment]
@PyLearner hi, pls update the commit log for more details

**Comment by vivianQizhu** [comment]
All blk_virtio_blk cases Passed, but all blk_virtio_scsi cases Failed:

14:59:05 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-38.0-py2.7.egg/avocado_vt/test.py:422
14:59:05 ERROR| Traceback (most recent call last):
14:59:05 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-38.0-py2.7.egg/avocado_vt/test.py", line 207, in runTest
14:59:05 ERROR|     raise exceptions.TestFail(details)
14:59:05 ERROR| TestFail: Check for block device failed after PCI hotplug. Output: 'echo 1; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[v<; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[vs                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08]d[a-z]" | cut -d" " -f2 <>&1 | grep "Disk /dev/[vs]d[a-z]" | cut -d" " -f2 |                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 cut -d: -f1 | sort | tai<d[a-z]" | cut -d" " -f2 | cut -d: -f1 | sort | tail                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 -1`\n1\nmke2fs 1.42.9 (28-Dec-2013)\n/dev/vda is apparently in use by the system; will not make a filesystem here!\n'    [context: Deleting SCSI controller.]
14:59:05 ERROR|
14:59:05 ERROR| FAIL 07-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.my-provider.pci_hotplug.hotplug_block.blk_virtio_scsi.with_shutdown -> TestFail: Check for block device failed after PCI hotplug. Output: 'echo 1; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[v<; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[vs                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08]d[a-z]" | cut -d" " -f2 <>&1 | grep "Disk /dev/[vs]d[a-z]" | cut -d" " -f2 |                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 cut -d: -f1 | sort | tai<d[a-z]" | cut -d" " -f2 | cut -d: -f1 | sort | tail                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 -1`\n1\nmke2fs 1.42.9 (28-Dec-2013)\n/dev/vda is apparently in use by the system; will not make a filesystem here!\n'    [context: Deleting SCSI controller.]

**Comment by PyLearner** [comment]
@vivianQizhu   thanks for you comment, i have renewd the patch ,could you help to review ,thanks.

 (1/1) smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_block.blk_virtio_scsi.with_block_resize: PASS

**Comment by vivianQizhu** [comment]
@PyLearner I still hit same issue with your update.
13:38:59 ERROR| FAIL 1-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.my-provider.pci_hotplug.hotplug_block.blk_virtio_scsi.with_reboot -> TestFail: Check for block device failed after PCI hotplug. Output: 'echo 1; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[v<; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[vs                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08]d[a-z]" | cut -d" " -f2 <>&1 | grep "Disk /dev/[vs]d[a-z]" | cut -d" " -f2 |                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 cut -d: -f1 | sort | tai<d[a-z]" | cut -d" " -f2 | cut -d: -f1 | sort | tail                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08 -1`\n1\nmke2fs 1.42.9 (28-Dec-2013)\n/dev/vda is apparently in use by the system; will not make a filesystem here!\n'    [context: Deleting SCSI controller.]

**Comment by PyLearner** [comment]
hi,  the cmd has been changed to: pci_test_cmd = "echo PCI_NUM; yes | mke2fs `fdisk -l 2>&1 | grep "Disk /dev/[vs]d[a-z]" | cut -d" " -f2 | cut -d: -f1 | tail -1`"

but still got "sort " from your error log  "cut -d: -f1 | sort | tail "  , so could you please check if you apply the new patch , or you could lend me your env ,thanks.

**Comment by vivianQizhu** [comment]
@PyLearner Sorry, my fault. 
14:37:11 INFO | PASS 1-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.my-provider.pci_hotplug.hotplug_block.blk_virtio_scsi.with_reboot

**Comment by vivianQizhu** [comment]
Acked-by: Qianqian Zhu qizhu@redhat.com

**Comment by CongLi** [comment]
@PyLearner 

How to handle the serial for Windows guest?

**Comment by PyLearner** [comment]
Hi @CongLi ,
    i tried windows guest ,it works fine ,
smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u3.raw.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.pci_hotplug.hotplug_block.blk_virtio_scsi.with_reboot: PASS

i will talk with you for more detail and get the meaning of you, thanks.

**Comment by CongLi** [comment]
No problem for me now.

Acked-by: Cong Li <coli@redhat.com>

**Comment by CongLi** [comment]
Thanks @PyLearner @vivianQizhu , merged.

**Pull request merged**